oml/functional/metrics.py

```diff
@@ -57,51 +61,30 @@ def calc_retrieval_metrics(
         metrics["precision"] = dict(zip(precision_top_k, precision))
 
     if map_top_k:
-        map = calc_map(gt_tops, n_gts, map_top_k)
-        metrics["map"] = dict(zip(map_top_k, map))
+        map_ = calc_map(gt_tops, n_gts, map_top_k)
+        metrics["map"] = dict(zip(map_top_k, map_))
+
+    if query_categories is not None:
+        metrics_cat = {c: take_unreduced_metrics_by_mask(metrics, query_categories == c) for c in query_categories}
```

Comment:
A bit strange to have a simple dict `metrics` but metrics keys are mixed with some categories keys on the same top level

```python
metrics["map"][5] = 0.5
metrics["cmc"][3] = 0.4
metrics["OVERALL"]["cmc"][3] = 0.4
metrics["cats"]["cmc"][3] = 0.1
metrics["pigs"]["cmc"][3] = 0.2
...
```

Reply:
no, it's:

```
{
        "cat": {"cmc": {1: 1.0}, "precision": {3: 2 / 3, 5: 2 / 3}},
        "dog": {"cmc": {1: 1.0}, "precision": {3: 1 / 2, 5: 1 / 2}},
        OVERALL_CATEGORIES_KEY: ...,
    }
```